src/ui/pages/Notifications/NotificationItem.tsx

```diff
@@ -18,66 +20,87 @@ const NotificationItem = ({
 }: {
   item: KeriaNotification;
   index: number;
-  handleNotificationClick: (value: string) => void;
+  handleNotificationClick: (item: KeriaNotification) => void;
 }) => {
   const connectionsCache = useAppSelector(getConnectionsCache);
+  const [notificationLabelText, setNotificationLabelText] =
+    useState<string>("");
+  const [loading, setLoading] = useState(true);
+  const connection = connectionsCache.filter(
+    (connection) => connection.id === item.connectionId
+  )[0]?.label;
+  const [multisigIcpDetails, setMultisigIcpDetails] =
+    useState<MultiSigIcpRequestDetails | null>(null);
 
-  const notificationLabel = (item: KeriaNotification) => {
-    const connection = connectionsCache.filter(
-      (connection) => connection.id === item.connectionId
-    )[0]?.label;
-    // TODO: Implement different credential types
-    const credential = i18n.t(
-      "notifications.tab.credentialtypes.driverslicence"
+  const getMultisigIcpDetails = async () => {
+    const details = await Agent.agent.multiSigs.getMultisigIcpDetails(
+      item.a.d as string
     );
+    setMultisigIcpDetails(details);
+  };
+
+  const fetchNotificationLabel = async () => {
+    const label = await notificationLabel(item);
+    setNotificationLabelText(label);
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    fetchNotificationLabel();
+  }, [item, multisigIcpDetails]);
+
+  const notificationLabel = async (item: KeriaNotification) => {
     switch (item.a.r) {
     case NotificationRoute.ExnIpexGrant:
       return i18next.t("notifications.tab.labels.exnipexgrant", {
         connection: connection,
-        credential: credential,
       });
     case NotificationRoute.MultiSigIcp:
+      await getMultisigIcpDetails();
       return i18next.t("notifications.tab.labels.multisigicp", {
-        connection: connection,
+        connection: multisigIcpDetails?.sender.label,
```

Comment:
For sure too slow to call `getMultisigIcpDetails` when in the list. Why can't we just use the `connection` ID and get the details via Redux as we had before?

Reply:
![08072024162623](https://github.com/cardano-foundation/cf-identity-wallet/assets/119612231/c18bb964-3869-4c79-a544-321ba80e0f1e)

Reply:
That's why 😂

Reply:
Yes, when the page is loaded, `multisigIcpDetails` will give you the initiator connection, any other members in the multisig and the threshold. You need all those details when it's loaded.

For the list you only really need the name of the connection that sent it. You can use the connection ID to get the connection details from Redux, this will have the label.

Reply:
Redux only contains connections that are not related to multi-sigs. I have created another ticket to improve on this so will resolve this